docs/TRANSCRIPTION.md

```diff
@@ -0,0 +1,222 @@
+
+Our endpoints will need the following information in the body:
+- `api/transcription/start.json` will need `channel`
+- `api/transcription/stop.json` will need `taskId` and `builderToken`
+
+If everything is executed properly, we send a successful response using our helper function with data we will fill in later.
+
+```ts
+import type { APIContext } from "astro";
+
+export async function POST({ request }: APIContext) {
+    const { channel } = await request.json()
+
+    if (!channel) {
+        return sendBadRequest("channel is required")
+    }
+
+    return sendSuccessfulResponse("<return data>")
+}
+```
+
+## Make Request Helper Functions
+The rest of this guide will focus on the implementation of Cloud Recording, which includes calling the Agora API. To simplify the code and have a uniform request structure, we will set up a helper function in `utils/makeRequest.ts`. 
+
+The `makeRequest` function defines all the headers, executes the request, and returns the response. 
+
+The inputs will need the `method` (we will use GET and POST), `url`, `body`, and the `credential`. 
+
+```ts
+export const makeRequest = async (method: string, url: string, credential: string, body?: string) => {
+    const headers = new Headers({
+        "Authorization": "basic " + credential,
+        "Content-Type": "application/json",
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": `${method}, OPTIONS`,
+    });
+
+    const res = await fetch(url, {
+        method: method,
+        headers: headers,
+        body: body
+    })
+    if (!res.ok) {
+        console.log(await res.text())
+        throw new Error("Failed to make request")
+    }
+    return res
+}
+```
+
+Another helper function in `utils/generateCredential.ts` will generate this credential. It is a base-64 encoded credential using the Customer ID and Customer Secret.
+
+```ts
+export const generateCredential = () => {
+const credential = import.meta.env.CUSTOMER_ID + ":" + import.meta.env.CUSTOMER_SECRET
+
+const base64_credential = btoa(credential)
+return base64_credential
+}
+```
+
+## Generate Transcription Resource
+The first step in transcribing your video calls is generating a real-time transcription resource. Behind the scenes, this spins up a backend Agora service that can transcribe video calls. 
+
+To generate this resource, we make our first request to the Agora service. This will use the endpoint `https://api.agora.io/v1/projects/${appId}/rtsc/speech-to-text/builderTokens` to create the resource and return a builderToken for that resource. We will use the channel name as an `instanceId` for this resource. 
+
+We don't need to create an endpoint for this function since it will only be used when starting a transcribing session on this same backend server. Because of this, we define this function in the `utils/generateResource.ts` file to keep the code organized. 
```

Comment:
This explanation is not clear. I understand what you are trying to say but to an outside developer it's a bit confusing. 

This is step one of two when starting a transcription, which only matters to the backend system initiating the RTT bot because its part of the url to make the start call itself.

Reply:
updated to:

This is step one of two when starting a transcription session. This step doesn't need to be exposed to our users because it is only used by the backend system to initiate the Speech-To-Text bot. Because of this, we define this function in the `utils/generateResource.ts` file to keep the code organized.